pyuvdata/utils.pyx

```diff
@@ -118,105 +112,110 @@ cpdef numpy.ndarray[dtype=numpy.int64_t] antnums_to_baseline(
+    # GPS positions PDF in docs/references folder
+    for ind in range(n_pts):
+        gps_p = sqrt(xyz[0, ind] ** 2 + xyz[1, ind] ** 2)
+        gps_theta = atan2(xyz[2, ind] * _gps_a, gps_p * _gps_b)
 
-    _lon[i] = atan2(xyz[i, 1], xyz[i, 0])
+        _lla[0, ind] = atan2(
+            xyz[2, ind] + _ep2 * _gps_b * sin(gps_theta) ** 3,
+            gps_p - _e2 * _gps_a * cos(gps_theta) ** 3,
+        )
 
-    gps_n = _gps_a / sqrt(1.0 - _e2 * sin(_lat[i]) ** 2)
-    _alt[i] = (gps_p / cos(_lat[i])) - gps_n
+        _lla[1, ind] = atan2(xyz[1, ind], xyz[0, ind])
 
-  return latitude, longitude, altitude
+        _lla[2, ind] = (gps_p / cos(lla[0, ind])) - _gps_a / sqrt(1.0 - _e2 * sin(lla[0, ind]) ** 2)
 
+    return lla
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
 @cython.cdivision(True)
-cpdef numpy.ndarray[dtype=numpy.float64_t] _xyz_from_latlonalt(
+cpdef numpy.ndarray[dtype=numpy.float64_t, ndim=2] _xyz_from_latlonalt(
     numpy.float64_t[::1] _lat,
     numpy.float64_t[::1] _lon,
     numpy.float64_t[::1] _alt,
 ):
-  cdef int n_pts = len(_lat)
-  cdef numpy.ndarray[dtype=numpy.float64_t, ndim=2] xyz = np.empty((n_pts, 3))
-  cdef numpy.float64_t _gps_a = gps_a
-  cdef numpy.float64_t _gps_b = gps_b
-  cdef numpy.float64_t _e2 = e_squared
-
-  # create a memoryview
+  cdef Py_ssize_t i
+  cdef int n_pts = _lat.shape[0]
+  cdef numpy.ndarray[dtype=numpy.float64_t, ndim=2] xyz = np.empty((3, n_pts), dtype=np.float64)
   cdef numpy.float64_t[:, ::1] _xyz = xyz
 
-  cdef numpy.float64_t gps_n
-  with nogil:
-    for i in range(n_pts):
-      gps_n = _gps_a / sqrt(1.0 - _e2 * sin(_lat[i]) ** 2)
+  cdef numpy.float64_t  sin_lat, cos_lat, sin_lon, cos_lon, gps_n
 
-      _xyz[i, 0] = (gps_n + _alt[i]) * cos(_lat[i]) * cos(_lon[i])
-      _xyz[i, 1] = (gps_n + _alt[i]) * cos(_lat[i]) * sin(_lon[i])
+  for ind in range(n_pts):
+      sin_lat = sin(_lat[ind])
+      sin_lon = sin(_lon[ind])
 
-      _xyz[i, 2] = (_gps_b ** 2 / _gps_a ** 2 * gps_n + _alt[i]) * sin(_lat[i])
+      cos_lat = cos(_lat[ind])
+      cos_lon = cos(_lon[ind])
 
-  return xyz.squeeze()
+      gps_n = _gps_a / sqrt(1.0 - _e2 * sin_lat ** 2)
+
+      _xyz[0, ind] = (gps_n + _alt[ind]) * cos_lat * cos_lon
+      _xyz[1, ind] = (gps_n + _alt[ind]) * cos_lat * sin_lon
+
+      _xyz[2, ind] = (b_div_a2 * gps_n + _alt[ind]) * sin_lat
+  return xyz
 
 # this function takes memoryviews as inputs
 # that is why _lat, _lon, and _alt are indexed below to get the 0th entry
 @cython.boundscheck(False)
 @cython.wraparound(False)
-cpdef numpy.ndarray[dtype=numpy.float64_t] _ENU_from_ECEF(
+cpdef numpy.ndarray[numpy.float64_t, ndim=2] _ENU_from_ECEF(
     numpy.float64_t[:, ::1] xyz,
     numpy.float64_t[::1] _lat,
     numpy.float64_t[::1] _lon,
     numpy.float64_t[::1] _alt,
 ):
-  cdef int i
-  cdef int nblts = xyz.shape[0]
-  cdef numpy.ndarray[dtype=numpy.float64_t, ndim=2] enu = np.empty((nblts, 3), dtype=np.float64)
+    cdef Py_ssize_t i
```

Comment:
I don't really care about indentation per se, but should this be 2 spaces instead of 4 to be consistent with the rest of the file?